### DISCUSSION TITLE
Lerna version without GIT tags in microservice monorepo

### DISCUSSION THREAD

**Discussion by benjhoo** [ghiscoding/lerna-lite#804] [Q&A] [answered]
Hi,
I'm discovering lerna-lite, which seems to match what I'm looking for to manage versioning and changelogs in my project.
But there is something that doesn't quite appeal to me : the creation of GIT tag for each version bump.

In fact, my project (nestJS) is set up as a monorepo and consists of :
- a main service (gateway)
- 18 independent microservices

When I execute the "version" command, lerna creates a tag for each of the microservices that have changed and undergone a version bump. But I don't really want to clutter my repo with an huge number of tags. 
I updated my lerna.json config file to add the option "noGitTagVersion: true". It works well, but I have to commit the release changes manually in that case.
I find unfortunate that it doesn't make the release commit despite that.
And I realize that if the "version" command doesn't generate GIT tags, the next executed "version" command is unable to determine which packages are affected by the new release and suggests bumping all the services.

How does the "version" command detect changes in packages ? 
Why does the "changed" command show "no changes" while the "version" command indicates that all my services will be bumped ?

Ideally, what I would like is that for a release:
- all services affected by changes have their version and changelo updated
- the overall project version number (root-level package.json) is also bumped and included in the release
- the release is committed

Can this be achieved in some way with lerna ?

**Comment by ghiscoding** [comment, marked as answer]
I'm not sure if you're using conventional changelog or not but you probably should since it will probably help with some of these questions. Personally I only ever used conventional changelog with a fixed version for all package, the version is calculated by `conventional-changelog-recommend-bump` and Lerna only uses it to update the packages and changelogs. I don't have enough knowledge of Lerna and Git to answer your questions and the goal of this project was mainly created for maintenance and to make it smaller by making all commands as optional, however I don't have full knowledge of how everything works, I just maintain it like I said. 

I would suggest you look at the original Lerna issues and/or ask questions on Stack Overflow which is probably the best thing to do in this case. Lerna-Lite is just a repackaged Lerna (except that the newer Lerna is becoming another Nx product and Lerna-Lite is not). 

> the overall project version number (root-level package.json) is also bumped and included in the release

We never include the root level package in the release, the only thing that can be updated is the root changelog (requires conventional changelog) but that only happens when you're using fixed version (aka not `independent`). Lerna-Lite itself uses fixed version for publishing, so its root changelog is always updated. When using fixed version, it only creates 1 tag, I'm not sure about independent mode since I don't use it. 

Searching on Lerna's GitHub, I found this [issue 2050](https://github.com/lerna/lerna/issues/2050)

**Reply by benjhoo** [reply]
Thank you for this complete answer.
I'm not favorable to the same fixed version for all my microservices because I'm using K8s for deploying. And that means that a new release, which could concern just 2 services, engages a full rebuild and deployment of all the microservices.

I will take a look to conventional-changelog, see what I can do with.
And probably ask my questions in parallel on Stackoverflow and/or Lerna discussions.

Thank you !

**Reply by ghiscoding** [reply]
ah yeah for sure, I'm just emphasizing fixed version because I have more experiences with it but indeed for your use case then `independent` mode is probably better for you. I don't know what K8 is, so I can't help with that either.

**Reply by benjhoo** [reply]
It's a short form for Kubernetes ;-)
